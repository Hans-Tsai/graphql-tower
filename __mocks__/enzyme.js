import { configure, ShallowWrapper } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

// https://github.com/airbnb/enzyme/issues/1295
ShallowWrapper.prototype.diveTo = function diveTo(name, options) {
  let component = this;

  while (component.name() !== name) {
    component = component.dive(options);
  }
  component = component.dive();

  return component;
};

export * from 'enzyme';
