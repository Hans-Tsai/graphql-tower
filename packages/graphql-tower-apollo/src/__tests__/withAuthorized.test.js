import fp from 'lodash/fp';
import React from 'react';
import { shallow } from 'enzyme';
import { withAuthorized } from '../';

jest.useFakeTimers();

window = {};

describe('withAuthorized', () => {
  it('no authorization', async () => {
    const App = fp.compose(
      withAuthorized({ redirect: '/login' }),
    )(() => <div />);

    window.location = '';

    const app = shallow(<App />, { context: { client: {} } });

    expect(app.diveTo('WithAuthorized(Component)')).toMatchSnapshot();
    expect(window.location).toBe('/login');
  });

  it('disabled', async () => {
    const App = fp.compose(
      withAuthorized({ redirect: '/login', disabled: true }),
    )(() => <div />);

    window.location = '';

    const app = shallow(<App />, { context: { client: {} } });

    expect(app.diveTo('WithAuthorized(Component)')).toMatchSnapshot();
    expect(window.location).toBe('');
  });

  it('authorized', async () => {
    const App = fp.compose(
      withAuthorized({ redirect: '/login' }),
    )(() => (<div />));

    window.location = '';

    const app = shallow(<App />, { context: { client: { authorized: true } } });
    expect(app.diveTo('WithAuthorized(Component)')).toMatchSnapshot();
    expect(window.location).toBe('');
  });

  it('when redirect is null', () => {
    const App = fp.compose(
      withAuthorized(),
    )(() => <div />);

    const app = shallow(<App />, { context: { client: {} } });

    expect(() => app.diveTo('WithAuthorized(Component)'))
      .toThrowError(new TypeError('redirect is required'));
  });
});
