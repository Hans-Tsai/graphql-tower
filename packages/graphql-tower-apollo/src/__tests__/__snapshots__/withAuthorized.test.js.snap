// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withAuthorized authorized 1`] = `
<Component
  client={
    Object {
      "authorized": true,
    }
  }
/>
`;

exports[`withAuthorized disabled 1`] = `
<Component
  client={Object {}}
/>
`;

exports[`withAuthorized no authorization 1`] = `""`;
